### PR TITLE
Add ADBC Bikeshare, Abu Dhabi

### PR DESCRIPTION
**Pull request by BenSto**
Feel free to review. This is a small Bikesharing System in Abu Dhabi. Thank you very much!

### CODE REVIEW

File: pybikes/adbc.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+class AdbcbikeshareStation(BikeShareStation):
+    def __init__(self, name, latitude, longitude, bikes, free, extra):
+        super(AdbcbikeshareStation, self).__init__()
+
+        self.name       = name
+        self.latitude   = float(latitude)
+        self.longitude  = float(longitude)
+        self.bikes      = int(bikes)
+        self.free       = int(free)
+        self.extra      = extra
```
Comment (eskerda) [inline pybikes/adbc.py]: There's no need anymore to build a custom station class if it's not necessarily needed. BikeShareStation constructor accepts name, latitude, longitude, etc parameters.

float and int casts must be done outside the constructor, though.

**Comment by eskerda** [comment]
From the JS source, this is a mapping of what fields on the feed mean:

```
ed.MapStation = function(stationJson) {
        this.id = stationJson.id;
        this.stationName = stationJson.s;
        this.terminalNumber = stationJson.n;
        this.isManuallyBlocked = stationJson.m;
        this.isSchemeSuspended = stationJson.su;
        this.status = stationJson.st;
        this.isBlocked = stationJson.b;
        this.lastUpdatedTimestamp = stationJson.lu;
        this.hasBikeKeyDispenser = stationJson.bk;
        this.isBikeKeyAvailable = stationJson.bl;
        this.latitude = stationJson.la;
        this.longitude = stationJson.lo;
        this.docksAvailable = stationJson.da;
        this.docksUnavailable = stationJson.dx;
        this.bikesAvailable = stationJson.ba;
        this.bikesUnavailable = stationJson.bx;
        if (stationJson.hvm) {
            this.hvm = stationJson.hvm
        }
    };
```

**Comment by eskerda** [comment]
* It's not clear that `bikesAvailable + docksAvailable == slots`: probably `.da + .dx + .ba + .bx` would actually mean total slots. It's not a mandatory field, so if it's not explicitly there we can skip it too.
* `status: 'online' if item['st'] == 1 else 'offline' `
* As far as we can, it's good to have unique identifiers.
  * `.n (aka terminalNumber)` is giving the same as `.s (aka stationName)`, so not this
  * there's a `.id` that we can use, and add it to extra as `'uid': str(item['id'])`
* There seems to be a "key dispenser" of some kind. This is something someone wanting to build an app just for this schema would appreciate:
  * `'has_bike_keys': item['bk'] and item['bl']`

**Comment by BenSto** [comment]
Thanks a lot for your feedback. I made a commit, regarding your comments.

**Comment by eskerda** [comment]
Thank you!!

:bike: :bike: :bike: :bike: :bike: :bike: :bike: :bike: :bike: :bike: :bike:

**Pull request merged**